### ISSUE TITLE
Infinite recursion in update(Model *model, Drawable *drawable)

### ISSUE THREAD

**Issue by kirill77** [LiangliangNan/Easy3D#75]
In the buffers.cpp, this function:

        void update(Model *model, Drawable *drawable)

Will go into infinite recursion if one supplies model which is PointCloud and Drawable which is Drawable::DT_TRIANGLES.

**Comment by LiangliangNan** [comment]
What is the purpose of a TrianglesDrawable for a PointCloud?

**Comment by kirill77** [comment]
It's my opinion that software has to be working in such a way that it
should handle even incorrect usage gracefully - return some kind of error
or some such instead of crashing. You don't agree?

**Comment by LiangliangNan** [comment]
Fully understanding the user intention can help better inform the user of the misuse of a function, in a more informative way.

**Comment by kirill77** [comment]
Got it. Well, I was using LinesDrawable to visualize points (each point was
represented as a small piece of line), but then I thought it would be
better to visualize my points as solid spheres. So I created
TrianglesDrawable instead of LinesDrawable. The rest of the code stayed as
it was though - so the code wasn't actually creating triangles and only had
points. I guess I hoped that it would represent my points as solid spheres
or something like that. I started the program and it wasn't working so I
had to step inside to see what's going on. Now for each point I create a
small pyramid consisting of 4 triangles and it's working, but it would be
good if the code was at least asserting instead of going into infinite
recursion on incorrect use.
